## Задача 1
### На складе лежат разные фрукты в разном количестве.
### Нужно написать функцию, которая на вход принимает любое количество названий фруктов и их количество, а возвращает общее количество фруктов на складе

In [11]:
d = dict(apple=35, banana=30, pinneaple=45)
d

{'apple': 35, 'banana': 30, 'pinneaple': 45}

In [12]:
sum_fruits = sum(d.values())
sum_fruits

110

## Задача 2
### Дан список с затратами на рекламу. Но в данных есть ошибки, некоторые затраты имеют отрицательную величину. Удалите такие значения из списка и посчитайте суммарные затраты
### [100, 125, -90, 345, 655, -1, 0, 200]
### Используйте list comprehensions

In [16]:
l = [100, 125, -90, 345, 655, -1, 0, 200]
l_one = [n for n in l if n>0 == 0]
l_one

[100, 125, 345, 655, 200]

In [19]:
l_one_sum = sum(l_one)
l_one_sum

1425

## Задача 3
Даны два списка.
Дата покупки
['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']
Суммы покупок по датам
[1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]
4.1 Найдите, какая выручка у компании в ноябре
Используйте list comprehensions
4.2 Найдите выручку компании в зависимости от месяца
Для этого напишите функцию, которая на вход принимает список с датами и список с выручкой, а на выходе словарь, где ключи - это месяцы, а значения - это выручка.
Используйте аннотирование типов.

In [76]:
import pandas as pd


dt = ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']
sm = [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]

In [80]:
df = pd.DataFrame({'dt': ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09'],
'sm': [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]})


In [81]:
df = pd.DataFrame(sm, index=pd.to_datetime(dt))
print(df[df.index.month==11].sum()[0])

25098


In [83]:
import datetime


df = pd.DataFrame()
df['Date'] = [datetime.datetime.strptime(i, '%Y-%m-%d') for i in dt]
df['Sum'] = sm
df['Month'] = pd.DatetimeIndex(df['Date']).month
df.groupby('Month').sum()

Sum
Month       
9      25647
10     28645
11     25098
12     45452

## Выручка у компании в ноябре = 25098

## Выручка по месяцам:
Month  Sum   
**9      25647**
**10     28645**
**11     25098**
**12     45452**